pkg/asset/machines/libvirt/worker.go

```diff
@@ -2,60 +2,96 @@
+func NonMasterMachineSets(config *types.InstallConfig, pool *types.MachinePool, userDataSecret string) ([]clusterapi.MachineSet, error) {
+	if pool.Name == "master" {
+		return nil, fmt.Errorf("pool name master is not supported")
+	}
+
+	if config.Platform.Libvirt == nil {
+		return nil, fmt.Errorf("invalid inputs")
+	}
+	clustername := config.ObjectMeta.Name
+	platform := config.Platform.Libvirt
+	// FIXME: libvirt actuator does not support any options from machinepool.
+	// mpool := pool.Platform.Libvirt
 
-// WorkerMachineSetTmpl is template for worker machineset.
-var WorkerMachineSetTmpl = template.Must(template.New("worker-machineset").Parse(`
-apiVersion: cluster.k8s.io/v1alpha1
-kind: MachineSet
-metadata:
-  name: {{.ClusterName}}-worker-0
-  namespace: openshift-cluster-api
-  labels:
-    sigs.k8s.io/cluster-api-cluster: {{.ClusterName}}
-    sigs.k8s.io/cluster-api-machine-role: worker
-    sigs.k8s.io/cluster-api-machine-type: worker
-spec:
-  replicas: {{.Replicas}}
-  selector:
-    matchLabels:
-      sigs.k8s.io/cluster-api-machineset: {{.ClusterName}}-worker-0
-      sigs.k8s.io/cluster-api-cluster: {{.ClusterName}}
-      sigs.k8s.io/cluster-api-machine-role: worker
-      sigs.k8s.io/cluster-api-machine-type: worker
-  template:
-    metadata:
-      labels:
-        sigs.k8s.io/cluster-api-machineset: {{.ClusterName}}-worker-0
-        sigs.k8s.io/cluster-api-cluster: {{.ClusterName}}
-        sigs.k8s.io/cluster-api-machine-role: worker
-        sigs.k8s.io/cluster-api-machine-type: worker
-    spec:
-      providerConfig:
-        value:
-          apiVersion: libvirtproviderconfig.k8s.io/v1alpha1
-          kind: LibvirtMachineProviderConfig
-          domainMemory: 2048
-          domainVcpu: 2
-          ignKey: /var/lib/libvirt/images/worker.ign
-          volume:
-            poolName: default
-            baseVolumeID: /var/lib/libvirt/images/coreos_base
-          networkInterfaceName: {{.Platform.Network.Name}}
-          networkInterfaceAddress: {{.Platform.Network.IPRange}}
-          autostart: false
-          uri: {{.Platform.URI}}
-      versions:
-        kubelet: ""
-        controlPlane: ""
-`))
+	total := int64(0)
+	if pool.Replicas != nil {
+		total = *pool.Replicas
+	}
+
+	provider := libvirtprovider.LibvirtMachineProviderConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "libvirtproviderconfig.k8s.io/v1alpha1",
+			Kind:       "LibvirtMachineProviderConfig",
+		},
+		DomainMemory: 2048,
+		DomainVcpu:   2,
+		IgnKey:       "/var/lib/libvirt/images/worker.ign",
+		Volume: &libvirtprovider.Volume{
+			PoolName:     "default",
+			BaseVolumeID: "/var/lib/libvirt/images/coreos_base",
+		},
+		NetworkInterfaceName:    platform.Network.Name,
+		NetworkInterfaceAddress: platform.Network.IPRange,
+		Autostart:               false,
+		URI:                     platform.URI,
+	}
+	name := fmt.Sprintf("%s-%s-%d", clustername, pool.Name, 0)
```

Comment:
You inline a bunch of other `Sprintf` calls; I think we should inline this below as well for consistency.

Reply:
the name variable has multiple consumers that's why.

Reply:
> the name variable has multiple consumers that's why.

Oops, sorry.  I'd meant to comment on the `MasterMachines` instance, where there is only the one consumer:

```console
$ git grep name origin/pr/573 -- pkg/asset/machines/aws/master.go | grep -A10 idx
origin/pr/573:pkg/asset/machines/aws/master.go:		name := fmt.Sprintf("%s-%s-%d", clustername, pool.Name, idx)
origin/pr/573:pkg/asset/machines/aws/master.go:				Name:      name,
origin/pr/573:pkg/asset/machines/aws/master.go:					"sigs.k8s.io/cluster-api-cluster":      clustername,
```